### ISSUE TITLE
A-record gone after update with empty value

### ISSUE THREAD

**Issue by laurensbl** [ddclient/ddclient#648]
ddclient couldn't fetch an IP and subsequently updated the host with "" (empty value), effectively deleting the A-record at the DDNS provider.

Should this be possible?

Log message:
```
WARNING:  cannot connect to checkipv4.dedyn.io:443 socket: Connection reset by peer SSL connect attempt failed
WARNING:  found neither IPv4 nor IPv6 address
SUCCESS:  updating domain_and_tld_obfuscated.tld: good: IP address set to 

regards,
   ddclient@lb001 (version 3.10.0)
```
Subsequent log message seems to confirm it updated the host with an empty value (nothing):
```
WARNING:  skipping update of domain_and_tld_obfuscated.tld from <nothing> to correct_ip_(obfuscated).
WARNING:  last updated <never> but last attempt on Wed Apr  3 22:39:53 2024 failed.
WARNING:  Wait at least 5 minutes between update attempts.
```
I tested to see if ddclient could create an A-record, but it didn't work, Had to make the A-record manually at the web interface of the DDNS provider. So this behavior is quite damaging, as the A-record is absent until manual intervention can be performed to recreate the A-record at the DDNS provider.

**Comment by LenardHess** [comment]
Prior to v3.11.0 ddclient used to proceed with bogus updates after IP acquisition failed. (See 1715bd3)

**Comment by laurensbl** [comment]
Ah, OK. Let's hope this change will be implemented into Debian Bookworm soon, otherwise the risk of crippling DNS will be there for the remainder of Deb12's lifetime.

**Comment by beadon** [comment]
Different dynamic DNS providers will have different behavior if you set set the record to a null value.  It's strange behavior that a dynamic DNS provider chose to delete the record when receiving a null!

As of today, May 2026 Bookworm still does not have a 4.0.0 version, this means your best chance of an upgrade is using what you see on the releases page for this GitHub repo until a package gets released officially.

Given the age of this ticket I will be closing the issue, please reopen if there's something more to add.

**Comment by laurensbl** [comment]
> As of today, May 2026 Bookworm still does not have a 4.0.0 version, this means your best chance of an upgrade is using what you see on the releases page for this GitHub repo until a package gets released officially.

> Prior to v3.11.0 ddclient used to proceed with bogus updates after IP acquisition failed. (See https://github.com/ddclient/ddclient/commit/1715bd3a177b831ba21eab57b4d24e344d1d3976)

I've updated to Debian 13 Trixie by now, and have v3.11.2 now, so it should be fixed. Well, as extra check I implemented some tests in a pre-launch script, which works for me up until today. Still quite a disruptive behavior. Maybe you can contact the dDNS provider to smooth things out. It's deSEC (https://desec.io/).

**Comment by beadon** [comment]
@laurensbl I don't have an ongoing relationship with them, and they don't appear to be listed as an official provider in the documentation here.  If you are the "customer" you likely will have more weight in a support request.  If you like I would be happy to respond to support queries if the agent(s) could respond on this open issue ?

Thant being said, we COULD list them as an integrated supported provider in the README relatively easily.  Would you like that ?

**Comment by laurensbl** [comment]
> Thant being said, we COULD list them as an integrated supported provider in the README relatively easily. Would you like that ?

Well... I don't know if that's up to me. All I can say that I use their (d)DNS service for over three years now and it has never faltered, apart from this issue. It's free, open source, security minded, based in Europe (German, strong privacy laws there) and simple to use. I switched to their service because my paid (d)DNS provider couldn't get dnssec right, renewing Let's Encrypt certs kept failing with dnssec enabled. Enabling dnssec with deSEC did not disrupt the Let's Encrypt renewal process. Apart from the issue from this ticket all ddclient operations worked smooth and without errors, so it seems compatible.

TLDR: I can recommend them from personal experience. 

> @laurensbl I don't have an ongoing relationship with them, and they don't appear to be listed as an official provider in the documentation here. If you are the "customer" you likely will have more weight in a support request. If you like I would be happy to respond to support queries if the agent(s) could respond on this open issue ?

I'll open a support issue with deSEC and refer to this ticket. Hopefully you can collaborate to eliminate this issue.

**Comment by peterthomassen** [comment]
This is Peter from deSEC.

We've been using the `myip=` syntax to delete A/AAAA records for about 10 years, and our users started relying on it (see, for example, https://github.com/desec-io/desec-stack/pull/1141#issuecomment-3309943845). It therefore would be very difficult for us to change now. (For details and other API features, see https://desec.readthedocs.io/en/latest/dyndns/update-api.html.)

We used https://help.dyn.com/perform-update.html as an orientation when implementing the update API. Based on that, the obvious way to remove an IP address would have been to pass `offline=YES`. However, that's a service property, not an address property. In particular, it allows no distinction between IPv4 and IPv6 records.

In dyn.com's docs, the empty `myip=` parameter's behavior is not defined, but "the system's best guess will be used". We figured that it's a pretty good guess that a user sending such an update request wants to remove the particular record. This works separately for `myip`, `myipv4`, `myipv6`. If that wasn't the intended meaning, then it seems like the request shouldn't have been sent in the first place.

It may sometimes be the case that a client sends requests it doesn't mean to, but that seems like a client-side bug. (Note that even if we rejected such requests, this would be a request the client normally wouldn't want to send when it was due to a failure in determining the local IP address. Using the connection address as a best guess also doesn't seem safe, as the client might have intended to specify a different address that it was unable to determine.) To us, the best way forward, as usual, seemed to not second-guess what the client sends, but actually interpret it rather literally. Clients sending requests they don't want to send (or whose meaning they can't tell) should stop sending them.

We're very happy to see that this has been fixed in 3.11, and that issues determining the IP address will now lead to skipping the update requests. That's a great development. We've had support tickets about this maybe three times a year; they will now likely go away as folks update over the next few years.

We'll be happy to be added as a supported provider in the README. Does this require anything beyond adding the link there? (For example, I don't now if there's documentation of provider-specific idiosyncrasies etc.)

**Comment by beadon** [comment]
@peterthomassen — thanks for jumping in with context on the deletion behavior, that was genuinely helpful.

We've put together a proposed documentation update on branch [`docs/desec-provider`](https://github.com/beadon/ddclient/tree/docs/desec-provider) that includes:

- deSEC added to the supported services list in the README
- A deSEC-specific example added to the `dyndns2` protocol documentation, showing `server=update.dedyn.io` and token-based login
- `desec-ipv4` and `desec-ipv6` built-in web check entries (`checkipv4.dedyn.io` / `checkipv6.dedyn.io`) so users can write `webv4=desec-ipv4` to use deSEC's own IP detection
- A note about the empty `myip` deletion behavior, referencing the original DynDNS2 spec and clarifying that ddclient skips the update entirely when IP detection fails, so it shouldn't be triggered under normal operation

Before we raise a PR, we wanted to check if the wording looks accurate from your side — particularly the example config and the deletion behavior note. Happy to adjust anything before it goes in.

**Comment by peterthomassen** [comment]
This looks good to me! Minor corrections:
- The dynDNS update password is displayed after sign-up when a .dedyn.io domain name was create during account sign-up. Alternatively, create an API token later at https://desec.io/tokens.
- There is no login; users are identified through the update password (API token) alone.

**Comment by beadon** [comment]
@peterthomassen — thank you for the corrections. We've updated the example to remove the `login=` field (since users are identified by the update password alone) and clarified that the dynDNS update password is shown at sign-up for `.dedyn.io` domains, with the option to create an API token later at https://desec.io/tokens.

The documentation PR is now open: #906. Happy to adjust anything else before it merges.
